iap_mail_domain_blacklist/hooks.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2025 Quartile
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo.service import db
+from odoo.sql_db import db_connect
+
+from odoo.addons.iap.tools import iap_tools
+
+_logger = logging.getLogger(__name__)
+
+
+def update_mail_domain_blacklist(cr):
+    cr.execute(
+        """
+        SELECT EXISTS (
+            SELECT 1 FROM information_schema.tables
+            WHERE table_name = 'ir_config_parameter'
+        )
+        """
+    )
+    table_exists = cr.fetchone()[0]
+    if not table_exists:
+        return
+    cr.execute(
+        "SELECT value FROM ir_config_parameter WHERE key = %s",
+        ("iap_mail_domain_blacklist.mail_domain_blacklist",),
+    )
+    param_value = cr.fetchone()
+    if param_value and param_value[0]:
+        additional_domains = {domain.strip() for domain in param_value[0].split(",")}
+        iap_tools._MAIL_DOMAIN_BLACKLIST.update(additional_domains)
+
+
+def post_load_hook():
+    for db_name in db.list_dbs():
```

Comment:
@AungKoKoLin1997 Can you check if this is better, please.
```suggestion
    db_list = http.db_list()
    for db_name in db_list:
```

Reply:
No,` http.db_list()` is not better for this use case. It filters databases based on `dbfilter `and is intended for web requests inside controllers, which might exclude some databases. Since this is a backend operation that needs to update all databases, `db.db_list()` is the better choice.